hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/key/OMKeyRequest.java

```diff
@@ -470,12 +490,209 @@ protected static List<OzoneAcl> getAclsForDir(KeyArgs keyArgs,
       OzoneAclUtil.inheritDefaultAcls(acls, bucketInfo.getAcls(), DEFAULT);
     }
 
-    // add itself acls
+    // add acls from clients
     acls.addAll(OzoneAclUtil.fromProtobuf(keyArgs.getAclsList()));
-
+    acls = acls.stream().distinct().collect(Collectors.toList());
     return acls;
   }
 
+  static long getMaxNumOfRecursiveDirs() {
+    return MAX_NUM_OF_RECURSIVE_DIRS;
+  }
```

Comment:
`getMaxNumOfRecursiveDirs()` is not required, `MAX_NUM_OF_RECURSIVE_DIRS` can be accessed directly now.